pennylane/measurements/measurements.py

```diff
@@ -216,58 +215,11 @@ def shape(self, device=None):
             QuantumFunctionError: the return type of the measurement process is
                 unrecognized and cannot deduce the numeric type
         """
-        if qml.active_return():
-            return self._shape_new(device=device)
-
-        shape = None
-
-        # First: prepare the shape for return types that do not require a
-        # device
-        if self.return_type in (Expectation, MutualInfo, Variance, VnEntropy):
-            shape = (1,)
-
-        if self.return_type == State and self.wires:
-            dim = 2 ** len(self.wires)
-            shape = (1, dim, dim)
-
-        # Determine shape if device with shot vector
-        if device is not None and device._shot_vector is not None:
-            shape = self._shot_vector_shape(device, main_shape=shape)
-
-        # If we have a shape, return it here
-        if shape is not None:
-            return shape
-
-        # Then: handle return types that require a device; no shot vector
-        if device is None and self.return_type in (Probability, State, Sample):
-            raise MeasurementShapeError(
-                "The device argument is required to obtain the shape of the measurement process; "
-                + f"got return type {self.return_type}."
-            )
-
-        if self.return_type == Probability:
-            len_wires = len(self.wires)
-            dim = self._get_num_basis_states(len_wires, device)
-            return (1, dim)
-
-        if self.return_type == State:
-
-            # Note: qml.density_matrix has its shape defined, so we're handling
-            # the qml.state case; acts on all device wires
-            dim = 2 ** len(device.wires)
-            return (1, dim)
-
-        if self.return_type == Sample:
-            len_wires = len(device.wires)
-            # qml.sample(some_observable) case  // qml.sample() case
-            return (1, device.shots) if self.obs is not None else (1, device.shots, len_wires)
-
         raise qml.QuantumFunctionError(
             "Cannot deduce the shape of the measurement process with unrecognized return_type "
             + f"{self.return_type}."
         )
```

Comment:
Minor: can we check if `qml.active_return()` is `True` before raising this error, and call `self._shape_new` if so? The reason being we might have new measurement processes that are added during the deprecation cycle for the new return types, and we want developers to implement `_shape_new` without implementing `shape`.

Reply:
Done!